website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -141,6 +141,48 @@ new S3Bucket(this, "moved-bucket", {
 
 This allows you to efficiently integrate existing resources into a `foreach` composition without destroying the existing deployed resource.
 
+### Move By Resource Address
+
+In instances where the move target workflow does not easily fit your use case, resource moves can be performed by directly specifying the full resource address to be moved to/from.
+
+#### Move To
+
+```typescript
+new S3Bucket(this, "bucket-moved", {
+  bucket: "move-bucket",
+}).moveToId("aws_s3_bucket.bucket-new");
+
+new S3Bucket(this, "bucket-new", {
```

Comment:
Do I still need to specify the same bucket twice? I thought the new design allows us to "in-place" replace it. The aspect also seems to work this way without adding sth to the construct tree so either the example or aspect is wrong.

Reply:
In short, moving can be done "in-place" (in a certain sense of the phrase) when moving from a resource id but not in moving to a resource id. At least if I'm understanding your point correctly.

For `moveToId` we of course need the destination resource (or in other words the place we want the resource to be after the move) to be present in the config.

For `moveFromId` I've since realized that the convention of marking the original resource with `hasMoved` to be redundant. So, yes the move can be done "in-place"

CC my response to Ansgar asking if the original resource can just be removed:
```
Would be a completely valid way to do it. My main reasoning at the time for including it was to keep some semblance of consistency between all the ways of moving resources. Or in other words, it felt odd to have users delete resource definitions in this workflow where they wouldn't in others.

Must admit that I was thinking too narrowly, as the old resource will most likely be deleted anyways. So the use of hasMoved is redundant.

Going to just remove the convention 👍
```

On another note, added both `moveToId` and `moveFromId` to just cover all potential ways a move can be perform. In general I see these workflows as something to have in between the move target workflow and using escape hatches. So hopefully in the odd case that the move target way of doing things doesn't fit a use case, there is some native way of doing it.

Reply:
> we of course need the destination resource

I beg to differ, if you know the new id you can change the toTerraform function to put your resource under this name versus the friendlyUniqueId that is currently there.

Reply:
Ahh I see what you mean now. Had a similar thought during the initial implementation with the `rename` function. Would require some rework of how `TerraformElement` overrides ids.

We come to the problem where if we override the id, we can't get the original `friendlyUniqueId`– and if we access the `friendlyUniqueId`, we then can't override the id.

Per the following code:

```typescript
private _logicalIdOverride?: string;
 private _friendlyUniqueId?: string;

public get friendlyUniqueId() {
    if (!this._friendlyUniqueId) {
      if (this._logicalIdOverride) {
        this._friendlyUniqueId = this._logicalIdOverride;
      } else {
        this._friendlyUniqueId = this.cdktfStack.getLogicalId(this);
      }
    }
    return this._friendlyUniqueId;
  }

public overrideLogicalId(newLogicalId: string) {
    ok(
      !this._fqnToken,
      "Logical ID may not be overridden once .fqn has been requested. Make sure to override the id before passing the construct to other constructs."
    );
    this._logicalIdOverride = newLogicalId;
  }
```

In either case we are only able to get half the information needed for the moved block when doing such an operation in place, as it currently stands.